include/spirv/unified1/spirv.core.grammar.json

```diff
@@ -11880,6 +11880,32 @@
         }
       ]
     },
+    {
+      "category" : "ValueEnum",
+      "kind" : "HostAccessQualifier",
```

Comment:
These new enums for `HostAccessQualifier` and the enums for `InitializationModeQualifier` aren't getting added to the headers because the generation scripts don't know about them.  Are we OK with that?

Actually, I'm a little surprised the header generation script ran correctly because it has checks for unhandled "operand kinds":

```c++
        if (type == OperandNone) {
            std::cerr << "Unhandled operand kind found: " << operandKind << std::endl;
            exit(1);
        }
```

Did the generation script run correctly or were other changes needed?

Reply:
I took a look at this in a bit more detail because all of the automated checks are passing.  It turns out that this error is certainly occurring, though it isn't causing the CI jobs to fail for some reason.  We should fix this.

See: https://github.com/KhronosGroup/SPIRV-Headers/actions/runs/5423707347/jobs/9931446730?pr=356#step:7:15

```
Run cd tools/buildHeaders
dos2unix: converting file DebugInfo.h to Unix format...
dos2unix: converting file OpenCLDebugInfo100.h to Unix format...
dos2unix: converting file AMD_gcn_shader.h to Unix format...
dos2unix: converting file AMD_shader_ballot.h to Unix format...
dos2unix: converting file AMD_shader_explicit_vertex_parameter.h to Unix format...
dos2unix: converting file AMD_shader_trinary_minmax.h to Unix format...
dos2unix: converting file NonSemanticDebugPrintf.h to Unix format...
dos2unix: converting file NonSemanticClspvReflection.h to Unix format...
dos2unix: converting file NonSemanticDebugBreak.h to Unix format...
Unhandled operand kind found: InitializationModeQualifier
```

Reply:
@bashbaug thanks for review and investigation! What do you think, should I add these enums to the generation script or to add them somewhere in different place?

Reply:
> should I add these enums to the generation script

Yes please.  This shouldn't be too bad.  You can have a look at the recent cooperative matrix PR #355 as an example since it also added a few new types (see the changes in tools/buildHeaders).

Once this is done I would also recommend checking that a basic test case disassembles and assembles properly since updates may be needed to SPIRV-Tools also.

Thanks!

Reply:
Hi @bashbaug! I've updated the headers generation. However I'm not sure how to test SPIR-V tools, because we don't have implementation for these headers in llvm-spirv yet. The CI checks are passing, could it be enough? Thanks!

Reply:
Can you please check that SPIRV-Tools builds with these changes?  That would probably be the minimum to check, since the interaction between SPIRV-Headers and SPIRV-Tools is not checked via CI.

I'd recommend doing a light touch-test to see if a SPIR-V module using the new decorations assembles, disassembles, and validates properly also, though if this is broken it'll only affect the users of these extensions but not everyone using SPIR-V.  You could do this manually, say by adding the decoration to a small existing SPIR-V ASM file, if the tooling to generate SPIR-V modules using these extensions is not implemented yet.

Reply:
I tried building SPIRV-Tools with these header changes just to see if it'd work and I hit build errors.  In case it's helpful:

```
In file included from /home/bashbaug/git/spirv-tools/source/operand.cpp:36:
/home/bashbaug/git/spirv-tools/build/operand.kinds-unified1.inc:871:99: error: ‘SPV_OPERAND_TYPE_INITIALIZATION_MODE_QUALIFIER’ was not declared in this scope; did you mean ‘SPV_OPERAND_TYPE_OPTIONAL_ACCESS_QUALIFIER’?
  871 |   {"InitModeINTEL", 6147, 1, pygen_variable_caps_GlobalVariableFPGADecorationsINTEL, 0, nullptr, {SPV_OPERAND_TYPE_INITIALIZATION_MODE_QUALIFIER}, 0xffffffffu, 0xffffffffu},
      |                                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                   SPV_OPERAND_TYPE_OPTIONAL_ACCESS_QUALIFIER
/home/bashbaug/git/spirv-tools/build/operand.kinds-unified1.inc:873:96: error: ‘SPV_OPERAND_TYPE_HOST_ACCESS_QUALIFIER’ was not declared in this scope; did you mean ‘SPV_OPERAND_TYPE_ACCESS_QUALIFIER’?
  873 |   {"HostAccessINTEL", 6168, 1, pygen_variable_caps_GlobalVariableHostAccessINTEL, 0, nullptr, {SPV_OPERAND_TYPE_HOST_ACCESS_QUALIFIER, SPV_OPERAND_TYPE_LITERAL_STRING}, 0xffffffffu, 0xffffffffu},
      |                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                SPV_OPERAND_TYPE_ACCESS_QUALIFIER
```

Reply:
Thanks Ben! I've also reproduced it locally, and just forgot to upload my fixes to the Tools - you can see the PR in the reference now :) 
Now tools are buildable, local ctests pass, and the "basic" spirv with added capability and extension are going without problems through spirv-val, spirv-as and spirv-dis.